### ISSUE TITLE
Keras fails to run on this simplest 3-column dataset!!

### ISSUE THREAD

**Issue by Sukii** [fchollet/keras#6260]
z=f(x,y)=x+y
There are 3 columns in the dataset with x running from 1 to 1040 and y running from 1040 to 1. The value of z is a constant 1041. Deep learning with Keras doesn't work on this simplest possible dataset. It always returns zero accuracy and zero prediction. Why?

[deep-learning-with-plus.zip](https://github.com/fchollet/keras/files/923342/deep-learning-with-plus.zip)

**Comment by joelthchao** [comment]
1. Definition of accuracy is exactly match, but your network provide a float number. Therefore, it almost impossible to get a accuracy larger than 0.
2. Data is not normalized, which is not good for deep learning.

**Comment by Sukii** [comment]
Even if I ran with Normalized data, mterics=['mae'] etc.
but still the prediction is always zero.

Venkat

**Comment by joelthchao** [comment]
```python
X = dataset[:,0:2] / 1041
Y = dataset[:,2] / 1041
print(numpy.round(predictions * 1041))
# acc: 100.00%
```

**Comment by Sukii** [comment]
[cmd.txt](https://github.com/fchollet/keras/files/923387/cmd.txt)
I still get 0% and 0.0 prediction
[deep-learning-plus.zip](https://github.com/fchollet/keras/files/923388/deep-learning-plus.zip)

**Comment by Sukii** [comment]
Is it running fine in your version of keras?

**Comment by joelthchao** [comment]
Keras 2.0.3. 
1. You didn't normalize Y.
2. Normalizer is normalizing over feature dim, which is not suitable here.

**Comment by Sukii** [comment]
​In the new dataset Y=1.0 not 1401, so it is normailzed but I am not
getting anything different.
Where you able to successfully run the code?
If you have a revised dataset and code that works, then can you send it to
me?

Venkat

**Comment by joelthchao** [comment]
Just change 3 line as I mention above:
```python
from keras.models import Sequential
from keras.layers import Dense, Activation
from sklearn.preprocessing import Normalizer
import numpy
# fix random seed for reproducibility
seed = 10
numpy.random.seed(seed)
# load division dataset
dataset = numpy.loadtxt("plus.csv", delimiter=",")
# split into input (X) and output (Y) variables
X = dataset[:,0:2] / 1041
#print(X)
Y = dataset[:,2] / 1041
#print(Y)
# create model
model = Sequential([
                  Dense(10, input_dim=2, init='uniform', activation='relu'),
 #                 Dense(2, init='uniform', activation='relu'),
                  Dense(1, init='uniform', activation='relu'),
             ])
# Compile model
model.compile(loss='mse', optimizer='adam', metrics=['accuracy'], vebose=1)
# Fit the model
model.fit(X, Y, nb_epoch=100, batch_size=100)
# evaluate the model
scores = model.evaluate(X, Y)
print("%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))
# calculate predictions
predictions = model.predict(X)
# round predictions
print(numpy.round(predictions * 1041))
```

**Comment by Sukii** [comment]
Oops!
I just switched the backend to Theano and it works like magic, so it is a
bug in Tensorflow after all!!
Theano also reported a syntax error "vebose=2" which should be "verbose"
but in case "verbose" option is not allowed in compile function, but only
in the model fitting function.
I any case changing that also didn't fix the Tensorflow bug ;)
but Theano works just as it should.

**Comment by joelthchao** [comment]
Actually I use tensorflow backend, so I don't think it's a tf bug.

**Comment by Sukii** [comment]
Then it must be a bug in the version I am using which might have been fixed
later.
I use an official docker image that has tensorflow version 0.12.1.
If its not a bug in tensorflow, what we will it be?

**Comment by joelthchao** [comment]
As I said before, accuracy is not used in that way. No matter accuracy is 0% or 100%, it's expected. All you can do is inference offline, round it and calculate accuracy in a correct way.

**Comment by Sukii** [comment]
You missed the point completely!!
I have been using this docker instance with Tensorflow for running other
datasets and problems fairly successfully.
Tensorflow did not run at all for me for this problem with all the
predictions staying zero even if we use metric='['mae']" or anything else.
Theano worked fine for the same problem, why?
Anyway, I am at present trying to create a fairly simple unary and binary
algebraic functions to see how good deep learning is at predicting them.
Strangely I find that it s struggling, despite all the hype that it can
approximate any continuous function etc ;)
